File: leetcode/editor/cn/1464_数组中两元素的最大乘积_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when1464_thenSuccess(t *testing.T) {
	assert.Equal(t, 12, maxProduct([]int{3, 4, 5, 2}), "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func maxProduct(nums []int) int {
	m1, m2 := nums[0], nums[1]
	if m1 < m2 {
		m1, m2 = m2, m1
	}
	for _, n := range nums[2:] {
		if n > m1 {
			m1, m2 = n, m1
		} else if n > m2 {
			m2 = n
		}
	}
	return (m1 - 1) * (m2 - 1)
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/1464_数组中两元素的最大乘积_test.py
import unittest
from typing import List


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertTrue(True, "ShouldBeTrue")


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        m1, m2 = nums[0], nums[1]
        if m1 < m2:
            m1, m2 = m2, m1
        for n in nums[2:]:
            if n > m1:
                m1, m2 = n, m1
            elif n > m2:
                m2 = n
        return (m1 - 1) * (m2 - 1)


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
